### DISCUSSION TITLE
How to add a custom .kext into OC config while also enabling Secure Boot?

### DISCUSSION THREAD

**Discussion by farmerpaul** [osy86/HaC-Mini#1066] [Q&A] [answered]
Hey there,

After overcoming a few bumps in the road, I'm thrilled to have successfully turned my NUC into a Mac, so thank you for all your efforts here, @osy and the community alike. I'm so grateful this is even possible.

I understand that the Ventura version of [OpenIntelWireless](https://github.com/OpenIntelWireless) is still in alpha, which is why isn't packaged in HaCMini yet. However, I've discovered in their support chat that many people have had success using it, even if it's not 100% stable, and I wanted to give it a go myself.

Unfortunately, OpenIntelWireless's installation docs are pretty barebones, making the assumption that anyone trying to use these drivers should already understand how to edit `config.plist` to load a new `.kext`. After spending some time learning how to even access my EFI partition, let alone add the driver to the right spot in it with the right kernel version, etc., I then spent hours trying to troubleshoot why OC assumed my `config.plist` was corrupted on boot. Google sent me down a variety of rabbit holes that I didn't quite understand in trying to troubleshoot that issue.

In the end, I learned that this had to do with having enabled Secure Boot too early (surprise!). Something about the way encryption is done and a changed `config.plist` messing everything up, rendering my system unbootable. Through this process I learned how to recover a working EFI partition, etc., and I appreciate the hints scattered throughout HaCMini docs that helped to inform me of these things. Gaining a ton of knowledge here.

So now I'm at a point where I really want HaCMini to do its magic in enabling Secure Boot, but I also want to have this new driver installed in a way that it doesn't corrupt my ability to boot!

I have already tested the `AirportItlwm.kext` extension _without_ Secure Boot enabled, and I confirmed it works great; I'm sure it will be fine for my purposes even if it crashes once in a while (this isn't my primary machine so I'm not worried).

So I'm trying to figure out the order of steps to get the driver loaded, while also having Secure Boot enabled:

- I assume I have to run `HaCMini.pkg` when I enable Secure Boot, because it seems to do some kind of magic on my EFI partition that I have no idea how to do manually
- I also need to edit `config.plist` and add the kernel extension to `EFI/OC/Kexts`
- I presumably also have to do that `bless` step in macOS recovery the first time I enable Secure Boot
- Maybe I also have to check the "Regenerate unique identifiers" in `HaCMini.pkg`?

If anyone has some insight into getting this driver included, while playing nicely with Secure Boot, I'd be thrilled to learn how. :pray:

**Comment by farmerpaul** [comment, marked as answer]
Just thought I'd share a solution to my own question, yet again!

I realized the best way to support another .kext (also also make it work with Secure Boot) is to just patch the HaCMini installer, which wasn't hard at all (thanks @osy for creating an easy to follow project!). I've included this change a branch on a forked version of HaCMini, in case others would like to test out the alpha version of the Ventura `AirportItlwm.kext` on their HaC Mini:

https://github.com/farmerpaul/HaC-Mini/tree/airportitlwm-ventura-alpha

Initial tests show it's working as expected.

**Reply by osy** [reply]
Thanks, I’ll look into merging this in the next update.

**Comment by Anycubic** [comment]
@farmerpaul do you have a compiled  .pkg that I can test? In your fork there is any,  thanks!

**Reply by farmerpaul** [reply]
It's been a while; I might have a pkg, but I'm on holiday and won't be able to check for at least another week. Sorry!

**Reply by Anycubic** [reply]
Hi! Did you check if you have the pkg available? Thanks

**Reply by farmerpaul** [reply]
Heya, sorry for not getting back. I _think_ this is working .pkg file I built! Shared it on [Google Drive](https://drive.google.com/file/d/1C2z7WjsdP88jn5yCLo20b_qT9ojX-6Zg/view?usp=sharing)
